MonoidalCategories/gap/SymmetricCoclosedMonoidalCategoriesDerivedMethods.gi

```diff
@@ -0,0 +1,590 @@
+                ] );
+
+    return TensorProductToInternalCoHomAdjunctionMap(
+             tensor_product_b1_b2,
+             TensorProductOnObjects( int_cohom_a1_b1, int_cohom_a2_b2 ),
+             morphism );
+
+end : CategoryFilter := IsSymmetricCoclosedMonoidalCategory,
+      Description := "InternalCoHomTensorProductCompatibilityMorphismWithGivenObjects using associator, braiding and the CoEvaluation morphism" );
+
+##
+AddDerivationToCAP( CoDualityTensorProductCompatibilityMorphismWithGivenObjects,
+
+  function( new_source, object_1, object_2, new_range )
+    local morphism, unit, tensor_product_on_object_1_and_object_2;
+    
+    unit := TensorUnit( CapCategory( object_1 ) );
+    
+    tensor_product_on_object_1_and_object_2 := TensorProductOnObjects( object_1, object_2 );
+    
+    morphism := PreCompose( [
+                  IsomorphismFromCoDualToInternalCoHom( tensor_product_on_object_1_and_object_2 ),
+                  
+                  InternalCoHomOnMorphisms(
+                    LeftUnitorInverse( unit ),
+                    IdentityMorphism( tensor_product_on_object_1_and_object_2 ) ),
+                  
+                  InternalCoHomTensorProductCompatibilityMorphism( unit, unit, object_1, object_2 ),
+                  
+                  TensorProductOnMorphisms(
+                    IsomorphismFromInternalCoHomToCoDual( object_1 ),
+                    IsomorphismFromInternalCoHomToCoDual( object_2 ) )
+                ] );
+
+    return morphism;
+
+end : CategoryFilter := IsSymmetricCoclosedMonoidalCategory,
+      Description := "CoDualityTensorProductCompatibilityMorphismWithGivenObjects using left unitor, and compatibility of internal cohom and tensor product" );
+
+##
+AddDerivationToCAP( IsomorphismFromInternalCoHomToObjectWithGivenInternalCoHom,
+                  
+  function( object, internal_cohom )
+    local unit;
+    
+    unit := TensorUnit( CapCategory( object ) );
+    
+    return PreCompose(
+             InternalCoHomOnMorphisms(
+               LeftUnitorInverse( object ),
+               IdentityMorphism( unit ) ),
+             DualCoEvaluationMorphism( unit, object ));
+    
+end : CategoryFilter := IsSymmetricCoclosedMonoidalCategory,
+      Description := "IsomorphismFromInternalCoHomToObjectWithGivenInternalCoHom using the dualcoevaluation morphism" );
+
+##
+AddDerivationToCAP( IsomorphismFromInternalCoHomToObjectWithGivenInternalCoHom,
+                  
+  function( object, internal_cohom )
+    local unit;
+    
+    unit := TensorUnit( CapCategory( object ) );
+    
+    return TensorProductToInternalCoHomAdjunctionMap(
+             unit,
+             object,
+             LeftUnitorInverse( object ) );
+    
+end : CategoryFilter := IsSymmetricCoclosedMonoidalCategory,
+      Description := "IsomorphismFromInternalCoHomToObjectWithGivenInternalCoHom as the adjoint of the left inverse unitor" );
+
+## TODO: modify for the coclosed case and enable
```

Comment:
What about this TODO (and the others further down)?

Reply:
These are TODO's I copied from the symmetric closed case (see [here](https://github.com/homalg-project/CAP_project/blob/4196a95cb3b74719fe1860d09dfb3bde5fde83cf/MonoidalCategories/gap/SymmetricClosedMonoidalCategoriesDerivedMethods.gi#L277)). The code was commented out, so I didn't activate it, but I have forgotten to translate it to the coclosed case. I can translate that if you prefer.

Reply:
Ah okay. No, I was just curious, thanks.